module_5/oop_classes.py

```diff
@@ -0,0 +1,171 @@
+        :return: message if guessed or not
+        """
+        if randomed == int(guessed):
+            msg = "Yehoo! You're lucky today"
+        else:
+            msg = "Ohh, try another time"
+        return msg
+
+
+class PdqeNewsPaperSite(CommonMethods):
+    """
+    Class that gets information from user and puts it into txt file
+    """
+
+    def __init__(self, output_file):
+        self.output_file_name = output_file
+
+    news = News()
+    ads = Ads()
+    luck = CheckYourLuck()
+    common_methods = CommonMethods()
+
+    print("*" * 20 + f" Welcome to PDQE Newspaper site " + "*" * 20)
+    print("""\nPlease choose category for your message:
+            1 - News
+            2 - Advertising
+            3 - Check your luck""")
+
+    def get_category(self):
+        """
+        Method to get category for user's input
+        :return: 
+        """
+        val = input("\nEnter your value: ")
+        return val
+
+    def user_input(self):
+        """
+        Method to populate file with user's data
+        """
+        val = self.get_category()
+        if int(val) == 1:
+            title = f"\n" + "*" * 20 + f" News " + "*" * 20
+            self.news_text, self.news_city = self.news.get_news()
+            self.common_methods.populate_output_file(title,
+                                                     f"News: {self.news_text}",
+                                                     f"City: {self.news_city}",
+                                                     f"Publish date: {self.common_methods.get_current_date()}",
+                                                     f"{self.output_file_name}")
+
+        elif int(val) == 2:
+            title = f"\n" + "*" * 20 + f" Private Ads " + "*" * 20
+            self.ads_text, self.expiration_date = self.ads.get_ads()
+            self.common_methods.populate_output_file(title,
+                                                     f"Ads: {self.ads_text}",
+                                                     f"Expiration date: {self.expiration_date}",
+                                                     f"Days left: {self.ads.get_days_left(str(self.expiration_date))}",
+                                                     f"{self.output_file_name}")
+
+        else:
+            title = f"\n" + "*" * 20 + f" Check your luck " + "*" * 20
+            self.user_choise = self.luck.get_user_number()
+            self.randomed = self.luck.get_random_int()
+            self.common_methods.populate_output_file(title,
+                                                     f"Your choice: {self.user_choise}",
+                                                     f"System choice: {self.randomed}",
+                                                     f"Result: {self.luck.success_of_fail(self.user_choise, self.randomed)}",
+                                                     f"{self.output_file_name}")
+
+        print(f"Please find results in {self.output_file_name}")
+
+
```

Comment:
It would also be possible not to exit the program after a single use. And leave the opportunity to use it several times.

Reply:
fixed